### ISSUE TITLE
No Printer Attached Issue

### ISSUE THREAD

**Issue by Donovan4097** [bigtreetech/SKR-3#57]
Hello, I am having the no printer attched issue with my printer. I have an Ender 5 plus with an SKR 3 EZ and a TFT50 V3.0.1. I will provide my configuration.h below but I have followed the manual, and I have gone through every no printer attched post I can find. Nothing helps, my current settings for the tft are Serial Port = 1, Serial Port 2 = -1, Serial Port 3 = 3 as this is what the manual reccomends. I have tried every combination that I can think of but maybe I am missing something. I would appreciate some help. It wont let me upload the .h file so here is the .txt file for config
[Configuration.txt](https://github.com/bigtreetech/SKR-3/files/11302666/Configuration.txt)
[Configuration_adv.txt](https://github.com/bigtreetech/SKR-3/files/11302668/Configuration_adv.txt)

**Comment by radek8** [comment]
Do you have the same serial speed set?
Did you use the correct coneltor?
Have you checked the TFT cable to see if it is faulty?

**Comment by Donovan4097** [comment]
I used the TFT connector and I ohmed the cables, they are good. and yes I have the serial speed set to 115200 on the screen and marlin. Im using marlin 2.0.9.3 does this matter? I read something that says I may need marlin 2.1 or higher.

**Comment by radek8** [comment]
Yes 2.0.9.3 is an old version, but it should not prevent the display from communicating with the SKR.
At most, some functions will not be available.

**Comment by Donovan4097** [comment]
I just finished building marlin 2.1.2 for the skr 3 ez. I used a premade firmware from btt but marlin has a built in skr 3 ez board. lets see if it works

**Comment by Donovan4097** [comment]
Update: new firmware didn't solve anything. Still have no printer connected.

**Comment by Tomzrk** [comment]
Mine wasn't working either, i had the wrong envs selected in Platformio

[platformio]
src_dir = Marlin
boards_dir = buildroot/share/PlatformIO/boards
default_envs = STM32H743Vx_btt
include_dir = Marlin

Instead it should be

[platformio]
src_dir = Marlin
boards_dir = buildroot/share/PlatformIO/boards
default_envs = STM32H723Vx_btt
include_dir = Marlin

**Comment by Donovan4097** [comment]
But there is no such thing as STM32H723 only 743

**Comment by Tomzrk** [comment]
Are you using the newest marlin ?

**Comment by Donovan4097** [comment]
Yes Marlin-Bugfix 2.1.x

**Comment by Donovan4097** [comment]
![STM32H7](https://user-images.githubusercontent.com/127460139/235029172-a769d811-16ff-4c15-b446-9c524e00ae7b.PNG)

**Comment by Tomzrk** [comment]
First you need to fix the motherboard in configuration.h 
It should be for skr 3ez not regular 

#define MOTHERBOARD BOARD_BTT_SKR_V3_0_EZ

**Comment by Tomzrk** [comment]
Than you go to platformio and change the line i sent you and try to build the firmware and try it

**Comment by Donovan4097** [comment]
![Capture](https://user-images.githubusercontent.com/127460139/235029924-a504f59d-dbb6-40a0-bc7b-655f8647765f.PNG)

**Comment by Tomzrk** [comment]
Can you show me the valid names

**Comment by Donovan4097** [comment]
![111](https://user-images.githubusercontent.com/127460139/235030182-23e436b3-5263-4bfc-8f56-c9c59d9379bb.PNG)
Error: Unknown environment names 'STM32H723Vx_BTT'. Valid names are 'BIGTREE_GTR_V1_0, at90usb1286_dfu, FYSETC_E4, STM32F103VE_ZM3E4V2_USB, mks_skipr_v1_nobootloader, STM32G0B1RE_btt, mks_robin_lite3, STM32F103VE, STM32F103RE_creality_maple, STM32F103RE_creality, STM32F103CB_malyan_maple, STM32H743Vx_btt, teensy35, simulator_linux_debug, FYSETC_S6_8000, sanguino1284p, mks_robin_nano35, STM32F103RC_btt_512K_maple, STM32F103RC_btt, trigorilla_pro, jgaurora_a5s_a1_maple, at90usb1286_cdc, teensy36, Opulo_Lumen_REV3, mks_robin_e3p_maple, mks_monster8, STM32F103VC_ZM3E4_USB_maple, DUE, LERDGEX_usb_flash_drive, STM32F103RC_maple, mks_robin_nano35_maple, STM32F407VE_black, mks_monster8_usb_flash_drive, teensy41, BIGTREE_OCTOPUS_V1_USB, STM32F103CB_malyan, mks_robin_lite3_maple, mks_robin_nano_v1v2_usbmod, LERDGES, BIGTREE_OCTOPUS_PRO_V1_F429_USB, mks_robin_e3, LPC1768, SAMD51_grandcentral_m4, BIGTREE_SKR_2_F429_USB, STM32F103VE_longer, Artillery_Ruby, chitu_f103_maple, STM32F103RC_btt_USB_maple, FLYF407ZG, STM32F103RC_fysetc_maple, 
trigorilla_pro_disk, LERDGEK_usb_flash_drive, mks_eagle_usb_flash_drive, melzi_optiboot, rambo, STM32F103VE_GTM32, mks_robin_nano_v1_3_f4_usbmod, BIGTREE_OCTOPUS_V1_F407_USB, STM32F103VE_ZM3E4V2_USB_maple, FYSETC_SPIDER_KING407, mks_robin_nano_v1v2, PANDA_PI_V29, melzi_optimized, mks_robin_lite, mks_robin_e3p, Opulo_Lumen_REV4, FYSETC_S6, sanguino644p, BIGTREE_SKR_2, mks_robin_pro, STM32F103RET6_creality_maple, mega1280, mks_skipr_v1, mks_robin_maple, BTT_SKR_SE_BX, mks_robin_nano_v3_1_usb_flash_drive_msc, mega2560ext, sanguino1284p_optimized, mks_tinybee, mks_robin_lite_maple, mks_robin_nano_v3_usb_flash_drive, mks_ks_robin_nano_v3, STM32G0B1RE_btt_xfer, mega2560, mingda_mpx_arm_mini, STM32F103RC_ZM3E2_USB, STM32F103RE_btt_maple, malyan_M300, STM32F103RC_ZM3E2_USB_maple, MightyBoard1280, BIGTREE_SKR_2_F429_USB_debug, mks_robin_e3_maple, STM32F401RC_creality_stlink, simulator_macos_debug, jgaurora_a5s_a1, mks_robin_pro_maple, STM32F103RC_creality, STM32F401RC_creality_jlink, STM32F401RC_btt, esp32, mks_monster8_usb_flash_drive_msc, STM32F103RE_creality_xfer, STM32F103VE_longer_maple, simulator_windows, chitu_f103, BIGTREE_SKR_PRO_usb_flash_drive, STM32F401RC_creality, trigorilla_pro_maple, simulator_macos_release, mks_robin_mini, STM32F103RE_maple, linux_native, STM32F103RE_btt, STM32F103VE_GTM32_maple, STM32F446_tronxy, LERDGEK, STM32F103VC_ZM3E4_USB, mks_robin_nano_v1v2_maple, TH3D_EZBoard_V2_no_bootloader, STM32F103RC_fysetc, FLY_MINI, MightyBoard2560, SAMD21_minitronics20, STM32F103RET6_creality_xfer, chitu_v5_gpio_init_maple, BIGTREE_E3_RRF, mks_eagle_usb_flash_drive_msc, chitu_v5_gpio_init, PANDA, include_tree, melzi, FYSETC_CHEETAH_V20, flsun_hispeedv1, STM32F103RET6_creality, STM32F103RC_meeb_maple, STM32F103RC_btt_512K_USB_maple, FLY_MINI_maple, BIGTREE_BTT002, Anet_ET4_no_bootloader, STM32F103RC_meeb, STM32F103RC_creality_xfer, BIGTREE_BTT002_VET6, STM32F103ZE, STM32F103RE_btt_USB_maple, STM32F103RE_creality_smartPro, STM32F103RE_btt_USB, mks_robin_nano_v3_usb_flash_drive_msc, BIGTREE_SKR_PRO, simulator_linux_release, STM32F070RB_malyan, Anet_ET4_OpenBLT, STM32F103RC_btt_512K, STM32F070CB_malyan, BIGTREE_SKR_2_F429, LERDGEX, STM32F103RE, BIGTREE_OCTOPUS_PRO_V1_F429, STM32F103RE_creality_smartPro_maple, mks_robin_nano_v3_1_usb_flash_drive, DUE_USB, STM32F103VE_creality, FYSETC_F6, ARMED, ERYONE_ERY32_MINI_maple, LPC1769, teensy20, BIGTREE_SKR_2_USB_debug, BIGTREE_SKR_2_USB, DUE_archim, melzi_optiboot_optimized, BIGTREE_OCTOPUS_V1_F407, mks_robin_nano_v3_1, STM32F103RC_btt_USB, rumba32, LERDGES_usb_flash_drive, teensy31, BIGTREE_OCTOPUS_V1, mks_robin_nano_v1_3_f4, BTT_EBB42_V1_1_filament_extruder, mks_robin'

**Comment by Tomzrk** [comment]
Double check the number on the board chip

**Comment by Donovan4097** [comment]
Im about to jump off a bridge, downloaded bugfix for the 3rd time tonight and now 723 shows up

**Comment by Tomzrk** [comment]
Make sure your board chip end with 723 too mine does

**Comment by Donovan4097** [comment]
THank you for your patience but that was wild

**Comment by Tomzrk** [comment]
No worries, I been through the same thing just trying to help with my little knowledge

**Comment by Donovan4097** [comment]
TFT is now connecting to my skr 3 ez, you are a lifesaver. thank you for all of your help.

**Comment by Tomzrk** [comment]
You're welcome

**Comment by giri456** [comment]
hello i also have the same issue except that my chip matches the environment in platform .io i also cant get the marlin mode working

**Comment by radek8** [comment]
@giri456
Your description of the problem is too general.
Write what HW you have, describe exactly your problem.
Send the configuration files for SKR and TFT that you used.
Send photos of the cable connection between SKR and TFT

**Comment by giri456** [comment]
@radek8
Sorry about that, first time on the forums. 
I am using a skr 3 ez and trying to connect it to a tft24 v1.1 in both marlin and touchscreen mode

For the tft i used the files found [here](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware) I directly coppied and pasted the ones that were for the tft24. 

this is my marlin config converted to text file so i could upload here
[configuration_adv.txt](https://github.com/bigtreetech/SKR-3/files/13801404/configuration_adv.txt)
[configuration.txt](https://github.com/bigtreetech/SKR-3/files/13801405/configuration.txt)

**Comment by giri456** [comment]
Sorry forgot about the photo.
![IMG_4650](https://github.com/bigtreetech/SKR-3/assets/155290104/98e9459f-5e32-4e48-b437-700be2617245)
![IMG_4649](https://github.com/bigtreetech/SKR-3/assets/155290104/28a238b1-0c24-4997-9bd0-482db9e85361)

**Comment by radek8** [comment]
Also send the config.ini file, with your edit

**Comment by giri456** [comment]
i used the default config.ini that is on the big tree tech github for this screen

**Comment by radek8** [comment]
The default communication speed of the serial port is 115200 for TFT
Your Marlin configuration is 250000
You must have the same baud rate configured.
(for touch mode)

**Comment by radek8** [comment]
In Marlin, enable the option
define M115_GEOMETRY_REPORT

**Comment by giri456** [comment]
made the changes and did them on a fresh copy of marlin 
[configuration.txt](https://github.com/bigtreetech/SKR-3/files/13801698/configuration.txt)
[configuration_adv.txt](https://github.com/bigtreetech/SKR-3/files/13801699/configuration_adv.txt)

**Comment by giri456** [comment]
![IMG_4653](https://github.com/bigtreetech/SKR-3/assets/155290104/098b63d6-9495-4461-b32d-d7cbb08e68c2)
![IMG_4652](https://github.com/bigtreetech/SKR-3/assets/155290104/30622d4c-0c8b-44b0-b31d-cad030556b53)

**Comment by giri456** [comment]
Got a few more pictures of what happened in marlin mode
![IMG_4655](https://github.com/bigtreetech/SKR-3/assets/155290104/7fa6f980-326f-4033-aac6-f83304a536a2)
![IMG_4654](https://github.com/bigtreetech/SKR-3/assets/155290104/dd4a0f9a-42e0-43ef-a3ae-3de93c2b81d1)

**Comment by giri456** [comment]
@radek8 
#define ST7920_DELAY_1 DELAY_NS(300)
#define ST7920_DELAY_2 DELAY_NS(300)
#define ST7920_DELAY_3 DELAY_NS(300)

I also tried adding this into config.h still doesnt work

**Comment by giri456** [comment]
@radek8 i managed to get the touch mode to work i think it looks like this now

**Comment by giri456** [comment]
![IMG_4658](https://github.com/bigtreetech/SKR-3/assets/155290104/d03c09db-326c-45b5-9044-5156fdf8ead8)

**Comment by giri456** [comment]
But Marlin mode is still not working

**Comment by giri456** [comment]
![IMG_4654](https://github.com/bigtreetech/SKR-3/assets/155290104/3af9501a-ac16-484c-8edf-9994c7491d43)

**Comment by radek8** [comment]
How long are your EXP1 and EXP2 cables? It looks like electromagnetic interference on the cables

**Comment by radek8** [comment]
If Touch Mode works for you, you don't need Marlin Mode. 
I haven't used Marlin mode in over a year.

**Comment by giri456** [comment]
@radek8 
![IMG_4669](https://github.com/bigtreetech/SKR-3/assets/155290104/b0c7c8e1-3612-4872-be70-24f6a45d931e)
I have uploaded a picture of the length of the cable. To confirm I can use a sd card on the tft with just the touch mode cable?

**Comment by radek8** [comment]
>To confirm I can use a sd card on the tft with just the touch mode cable?

Yes, you can use the SD card on the TFT with a serial (5pin) cable.
If you set in Marlin:
#define SDCARD_CONNECTION ONBOARD
Then the mini SD card on the SKR will also be available in touch mode.

**Comment by giri456** [comment]
Ok thanks

**Comment by giri456** [comment]
@radek8 i finished my config.h and adv.h. i just tried flashing the firmware today and the file doesnt get renamed. the tft says eeprom data size error

**Comment by radek8** [comment]
Did you compile for the right platform? 
After updating the marlin, you should initialize the EEPROM

**Comment by giri456** [comment]
@radek8 I have compiled to the right processor in platform.io. But it still doesn’t rename the file. I also tried uploading it with only the .bin file and no .cur file. I also tried to rename the .bin file to full uppercase.

**Comment by radek8** [comment]
Are you updating Marlin?.
Use the microSD slot on the SKR.  
Use an SD card that is 8GB or smaller. 
How big is your BIN file? 
What is the designation of your processor on the SKR?

**Comment by giri456** [comment]
Turns out it was a sd card problem. But now after flashing my z2 stepper won’t move or lock. It’s like it’s not getting power.

**Comment by giri456** [comment]
[config_adv.h.txt](https://github.com/bigtreetech/SKR-3/files/13860893/config_adv.h.txt)
[config.h.txt](https://github.com/bigtreetech/SKR-3/files/13860894/config.h.txt)

**Comment by giri456** [comment]
nvm got everything working much thanks @radek8

**Comment by radek8** [comment]
I see that you don't have a USB port activated in Marlin. Are you going to use the USB port on the SKR? (On connection with Pc)

**Comment by BruceY19** [comment]
someone up above said only STM32H741 BOARDS around.   I have a SKR3EZ with STM32H723 chip
